### PR TITLE
[SPARK-43549][SQL] Convert _LEGACY_ERROR_TEMP_0036 to INVALID_SQL_SYNTAX.ANALYZE_TABLE_UNEXPECTED_NOSCAN

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to Convert `_LEGACY_ERROR_TEMP_0036` to INVALID_SQL_SYNTAX.ANALYZE_TABLE_UNEXPECTED_NOSCAN

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA & Update UT.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk 
Sorry, The previous PR was damaged by me, this is a new one.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1198,6 +1198,11 @@
       "Invalid SQL syntax:"
     ],
     "subClass" : {
+      "ANALYZE_TABLE_UNEXPECTED_NOSCAN" : {
+        "message" : [
+          "ANALYZE TABLE ... COMPUTE STATISTICS ... <ctx> must be 'NOSCAN'"
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1203]: 1. This error can come from slightly other command: `ANALYZE TABLES IN default COMPUTE STATISTICS`
2. Let's follow `toSQLStmt`, and don't wrap NOSCAN.

Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: ANALYZE TABLE ... COMPUTE STATISTICS ... <ctx> must be 'NOSCAN'
->
ANALYZE TABLE(S) ... COMPUTE STATISTICS ... <ctx> must be either NOSCAN or empty

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Pull request closed without merging**